main.py

```diff
@@ -298,37 +254,68 @@ def status(message):
     LOGGER.info("Hostname: %s", hostname)
 
     # Start reading from sensors
-    sensor_thread = Thread(target=read_data,
-                           args=(output_sensors, input_sensors, queue))
+    sensor_thread = Thread(target=read_data, args=(output_sensors, input_sensors, queue))
     sensor_thread.start()
-
-    # Start server
-    LOGGER.info("Starting server")
-    server = CoAPServer(("224.0.1.187", 5683), multicast=True)
-    server.add_resource('data/', DataResource(queue, input_sensors))
-    server.add_resource('name={}'.format(hostname), DummyResource())
-
+    
+    #load config file
+    try:
+        with open(config_file, 'r') as ymlfile:
+            cfg = yaml.load(ymlfile)
+    except:
+        LOGGER.exception("Error loading config file")
+
+    # Create mqtt client
+    client = paho.Client()
+    client.username_pw_set(username=cfg['mqtt']['uname'],password=cfg['mqtt']['password'])
+    #define callabcks
+    client.on_connect=on_connect
+    client.on_publish = on_publish
+    client.on_disconnect=on_disconnect
+    #reconnect interval on disconnect
+    client.reconnect_delay_set(3)
+    #establish client connection
     while True:
         try:
-            server.listen()
+            client.connect(cfg['mqtt']['server'],cfg['mqtt']['port'])
+            LOGGER.info("Client connected successfully to broker")
+            break
+        except:
+            LOGGER.exception("Connection failure...trying to reconnect...")
+            time.sleep(15)
+    client.loop_start()
+    #continuosly get data from queue and publish to broker
+    while True:
+        try:
+            data=queue.peek(blocking=True)
+            data={k.decode():(v.decode() if isinstance(v, bytes) else v, u.decode()) for k,(v,u) in data.items()}
+            data=json.dumps(data)
+            info=client.publish("prism/"+cfg['mqtt']['uname'],data, qos=1)
+            info.wait_for_publish()
+            while info.rc!=0:
+                time.sleep(10)
+                info=client.publish("prism/"+cfg['mqtt']['uname'],data , qos=1)
+                info.wait_for_publish()
+            LOGGER.info("Deleting data from queue")
+            queue.delete()
+            queue.flush()
```

Comment:
One thing I forgot: we need to update all of the input sensors that the data has been published (this will update the display on the sensor). Something like this:

```
for sensor in input_sensors:
    sensor.transmitted_data(len(queue))
```